### PR TITLE
add proposal filter feature (1.0)

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/190 for 1.0

Currently:

- Implements the feature on the DAO History page
- Changes the DAO History page to show _all_ proposals, not just IProposalStage.ExpiredInQueue, IProposalStage.Executed, IProposalStage.Queued and closingAt_lte: now
- adds DaoHistoryPage.scss
- consolidates the proposals query into one place (was in two)
- Behavior of search input is different from the DAOs search.  Executes on Enter and blur (see https://github.com/daostack/alchemy/issues/2103)
- is entirely case-sensitive
- does not search Description field.  Subgraph's GraphQL does not support 'or' across fields in a query.  This this would even require executing two queries, or would require that the user select in the UI what field they want to search by.  I would prefer to do that as a separate ticket and PR.  Same for searching through tags.  @Theylon
- there is a spinny icon that shows on long queries, but not when clearing the filter.

Heads up:
- when the filtered list is showing, the values will not automatically update on proposal state changes.  The DAOs list filter has the same behavior.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,8 +68,8 @@
     "start-prod": "cross-env NODE_ENV=production SHOW_ALL_DAOS=false NETWORK=main node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js  --config webpack.dev.config.js",
     "start-staging-kovan": "cross-env NODE_ENV=production SHOW_ALL_DAOS=true NETWORK=kovan node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --config webpack.dev.config.js",
     "start-staging-rinkeby": "cross-env NODE_ENV=production SHOW_ALL_DAOS=true NETWORK=rinkeby node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --config webpack.dev.config.js",
+    "start-staging-xdai": "cross-env NODE_ENV=production SHOW_ALL_DAOS=true NETWORK=xdai node --max_old_space_size=4096 ./node_modules/webpack-dev-server/bin/webpack-dev-server.js --config webpack.dev.config.js",
```
Comment (orenyodfat) [inline package.json:71]: is this related to this pr ?

Comment (dkent600) [reply package.json:71]: no, just a little harmless correction

File: src/components/Dao/DaoHistoryPage.tsx [hunk 2]
```diff
@@ -34,24 +82,41 @@ class DaoHistoryPage extends React.Component<IProps, null> {
     });
   }
 
+  onSearchChange = async (e: any) => {
```
Comment (roienatan) [inline src/components/Dao/DaoHistoryPage.tsx:85]: Is there a reason why this function is `async`?

**Comment by roienatan** [comment]
Is there a reason that the search is case-sensitive? Usually it's more convenient that it's not case-sensitive.

File: src/components/Dao/DaoHistoryPage.tsx (hunk 2 above)
Comment (dkent600) [reply src/components/Dao/DaoHistoryPage.tsx:85]: This is an artifact of a previous change, but since that has been deleted, we can remove the async.

**Comment by dkent600** [comment]
> Is there a reason that the search is case-sensitive? Usually it's more convenient that it's not case-sensitive.

Yes, thegraph doesn't support case insensitive searches.

**Comment by roienatan** [comment]
There are some indentations errors by Travis, please fix.

**Comment by dkent600** [comment]
> There are some indentations errors by Travis, please fix.

This is tricky because for some reason vscode insists on indenting in this way that eslint is rejecting.  I need to figure out why, so won't won't get to this until Monday.  Normally vscode matches eslint behavior perfectly.

**Review by orenyodfat — CHANGES_REQUESTED** [review]
@dkent600 there a lint issues

**Review by roienatan — APPROVED** [review]

**Review by orenyodfat — APPROVED** [review]
